File: solutions/clojure/167-k-closest-points-to-origin.clj
;; Problem 167: K Closest Points to Origin (LeetCode 973)
;; Difficulty: Med
;; Language: Clojure
;; 
(defn k-closest [points k]
  (->> points
       (sort-by (fn [[x y]] (+ (* x x) (* y y))))
       (take k)
       vec))

File: solutions/rust/167-k-closest-points-to-origin.rs
/**
 * Problem 167: K Closest Points to Origin (LeetCode 973)
 * Difficulty: Med
 * Language: Rust
 */
fn k_closest(points: &mut Vec<Vec<i32>>, k: usize) -> Vec<Vec<i32>> {
    points.sort_by_key(|p| p[0] * p[0] + p[1] * p[1]);
    points[..k].to_vec()
}
